test/mz-e2e/mzcompose.py

```diff
@@ -0,0 +1,254 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import argparse
+import csv
+import json
+import os
+import ssl
+import time
+import urllib.parse
+
+import pg8000
+
+from materialize.mzcompose import (
+    _wait_for_pg,
+)
+from materialize.mzcompose.composition import (
+    Composition,
+    WorkflowArgumentParser,
+)
+from materialize.mzcompose.services.cockroach import Cockroach
+from materialize.mzcompose.services.materialized import Materialized
+from materialize.mzcompose.services.mz import Mz
+from materialize.mzcompose.services.testdrive import Testdrive
+from materialize.ui import UIError
+
+REGION = "aws/us-east-1"
+ENVIRONMENT = os.getenv("ENVIRONMENT", "staging")
+USERNAME = os.getenv("NIGHTLY_MZ_USERNAME", "infra+bot@materialize.com")
+APP_PASSWORD = os.environ["MZ_CLI_APP_PASSWORD"]
+VERSION = "devel-" + os.environ["BUILDKITE_COMMIT"]
+
+SERVICES = [
+    Cockroach(setup_materialize=True),
+    Materialized(
+        # We use materialize/environmentd and not materialize/materialized here
+        # in order to ensure a perfect match to the container that should be
+        # deployed to the cloud
+        image=f"materialize/environmentd:{VERSION}",
+        external_cockroach=True,
+        persist_blob_url="file:///mzdata/persist/blob",
+        options=[
+            "--orchestrator-process-secrets-directory=/mzdata/secrets",
+            "--orchestrator-process-scratch-directory=/scratch",
+        ],
+    ),
+    Testdrive(),  # Overriden below
+    Mz(
+        region=REGION,
+        environment=ENVIRONMENT,
+        app_password=APP_PASSWORD,
+    ),
+]
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    """Deploy the current source to the cloud and run tests."""
+
+    parser.add_argument(
```

Comment:
I do not think you are using any of those command-line arguments, so consider removing them.